### DISCUSSION TITLE
passing VNC password from react-native to react.js running 'react-vnc'

### DISCUSSION THREAD

**Discussion by colinm** [roerohan/react-vnc#39] [Q&A] [answered]
Writing a react native vnc viewer for android that opens via webview a react.js app using react-vnc (on the react.js side) I want to pass a password via native to the instance running react-vnc in react.js. (this is running in a lan env and for a custom use case)

how could i edit this component to include the password passed?

 // other non relevant code above 
  const uri = params.get("host") + ":" + params.get("port");
  const pw = params.get("pass");
    <VncScreen
      url={uri}
      scaleViewport
      showDotCursor
      background="#000000"
      style={{
        width: "100vw",
        height: "100vh",
      }}
      ref={ref}
    />

**Comment by colinm** [comment, marked as answer]
This worked.

<VncScreen
      url={vncserver}
      scaleViewport
      rfbOptions={{
        credentials: {
          password: pw,
        },
      }}
      showDotCursor
      background="#000000"
      style={{
        width: "100vw",
        height: "100vh",
      }}
      ref={ref}
    />

**Reply by roerohan** [reply]
Yep, this is the right way to do it. As mentioned in the `README`, these is the `RFBOptions` type.

```ts
interface Props {
  url: string;
  style?: object;
  className?: string;
  viewOnly?: boolean;
  rfbOptions?: Partial<RFBOptions>;
  focusOnClick?: boolean;
  clipViewport?: boolean;
  dragViewport?: boolean;
  scaleViewport?: boolean;
  resizeSession?: boolean;
  showDotCursor?: boolean;
  background?: string;
  qualityLevel?: number;
  compressionLevel?: number;
  autoConnect?: number; // defaults to true
  retryDuration?: number; // in milliseconds
  debug?: boolean; // show logs in the console
  loadingUI?: React.ReactNode; // custom component that is displayed when loading
  onConnect?: (rfb?: RFB) => void;
  onDisconnect?: (rfb?: RFB) => void;
  onCredentialsRequired?: (rfb?: RFB) => void;
  onSecurityFailure?: (e?: { detail: { status: number, reason: string } }) => void;
  onClipboard?: (e?: { detail: { text: string } }) => void;
  onBell?: () => void;
  onDesktopName?: (e?: { detail: { name: string } }) => void;
  onCapabilities?: (e?: { detail: { capabilities: RFB["capabilities"] } }) => void;
}

// The rfbOptions object in Props is of type Partial<RFBOptions>
interface RFBOptions {
    shared: boolean;
    credentials: {
        username?: string;
        password?: string;
        target?: string;
    };
    repeaterID: string;
    wsProtocols: string;
}
```
